### ISSUE TITLE
v6: build error on test

### ISSUE THREAD

**Issue by bsr203** [erikras/redux-form#971]
just cloned the v6 branch.

```
git clone -b v6 --single-branch https://github.com/erikras/redux-form    
```

and `npm install` gives error  (`npm run build` is fine)

```
npm i                                                                                                                                                      v6
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.

> fsevents@1.0.12 install /Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/fsevents/lib/binding/Release/node-v47-darwin-x64/fse.node" is installed via remote

> redux-form@6.0.0-alpha.10 prepublish /Users/bsr/Work/Programming/javascript/repo/redux-form
> npm run test && npm run clean && npm run build


> redux-form@6.0.0-alpha.10 test /Users/bsr/Work/Programming/javascript/repo/redux-form
> mocha --compilers js:babel-register --recursive --recursive 'src/**/__tests__/*' --require src/__tests__/setup.js

/Users/bsr/Work/Programming/javascript/repo/redux-form/src/__tests__/setup.js:11
global.navigator.product = 'NotReactNative';
                         ^

TypeError: Cannot set property product of [object Object] which has only a getter
    at Object.<anonymous> (setup.js:7:1)
    at Module._compile (module.js:413:34)
    at loader (/Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/babel-register/lib/node.js:158:5)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/babel-register/lib/node.js:168:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at /Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/mocha/bin/_mocha:310:3
    at Array.forEach (native)
    at Object.<anonymous> (/Users/bsr/Work/Programming/javascript/repo/redux-form/node_modules/mocha/bin/_mocha:309:10)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:141:18)
    at node.js:933:3

npm ERR! Darwin 13.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "test"
npm ERR! node v5.7.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE
npm ERR! redux-form@6.0.0-alpha.10 test: `mocha --compilers js:babel-register --recursive --recursive 'src/**/__tests__/*' --require src/__tests__/setup.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the redux-form@6.0.0-alpha.10 test script 'mocha --compilers js:babel-register --recursive --recursive 'src/**/__tests__/*' --require src/__tests__/setup.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the redux-form package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     mocha --compilers js:babel-register --recursive --recursive 'src/**/__tests__/*' --require src/__tests__/setup.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs redux-form
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls redux-form
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/bsr/Work/Programming/javascript/repo/redux-form/npm-debug.log

npm ERR! Darwin 13.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i"
npm ERR! node v5.7.0
npm ERR! npm  v3.6.0
npm ERR! code ELIFECYCLE
npm ERR! redux-form@6.0.0-alpha.10 prepublish: `npm run test && npm run clean && npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the redux-form@6.0.0-alpha.10 prepublish script 'npm run test && npm run clean && npm run build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the redux-form package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run test && npm run clean && npm run build
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs redux-form
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls redux-form
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/bsr/Work/Programming/javascript/repo/redux-form/npm-debug.log

````

**Comment by bsr203** [comment]
offending line was 

```
global.navigator.product = 'NotReactNative' in setup.js
```
  getDisplayName
    ✓ should read displayName
    ✓ should read name
    ✓ should default to Component


  501 passing (5s)

~/W/P/j/r/redux-form ❯❯❯ git diff                                                                                                                                                 v6 ✱
diff --git a/src/__tests__/setup.js b/src/__tests__/setup.js
index 9c2d05b..71f5ea4 100644
--- a/src/__tests__/setup.js
+++ b/src/__tests__/setup.js
@@ -4,4 +4,4 @@ import { jsdom } from 'jsdom'
 global.document = jsdom('<!doctype html><html><body></body></html>')
 global.window = document.defaultView
 global.navigator = global.window.navigator
-global.navigator.product = 'NotReactNative'
+// global.navigator.product = 'NotReactNative'
~/W/P/j/r/redux-form ❯❯❯

**Comment by ooflorent** [comment]
This is strange. I've recently setup the v6 dev environment and didn't experienced any issue. Did you manage to this fix?

My dev env is a bit outdated but I'm using:

* node@5.9.0
* npm@3.7.3

**Comment by ooflorent** [comment]
This is probably related to a recent update of `jsdom`. v9.1.0 was released a few hours before you opened this issue and they reimplemented how `navigator` works.

As a temporary workaround you could force the version of `jsdom` to be `9.0.*`:

```
npm install jsdom@9.0
```

We should update the tests to fix how we use `navigator`.

**Comment by erikras** [comment]
Fixed by #977.
